openid-4-verifiable-presentations-1_0.md

```diff
@@ -417,7 +417,7 @@ This specification defines the following values for the `client_id_scheme` param
 
 * `pre-registered`: This value represents the [@!RFC6749] default behavior, i.e., the Client Identifier needs to be known to the Wallet in advance of the Authorization Request. The Verifier metadata is obtained using [@!RFC7591] or through out-of-band mechanisms.
 
-* `redirect_uri`: This value indicates that the Verifier's redirect URI is also the value of the Client Identifier. In this case, the Authorization Request MUST NOT be signed, the Verifier MAY omit the `redirect_uri` Authorization Request parameter, and all Verifier metadata parameters MUST be passed using the `client_metadata` or `client_metadata_uri` parameter defined in (#vp_token_request). 
+* `redirect_uri`: This value indicates that the Verifier's redirect URI is also the value of the Client Identifier. In this case, the Authorization Request MUST NOT be signed. The Verifier MAY omit the `redirect_uri` Authorization Request parameter. All Verifier metadata parameters MUST be passed using the `client_metadata` or `client_metadata_uri` parameter defined in (#vp_token_request). When Response Mode `direct_post` is used, these restrictions apply to the `response_uri` value instead of `redirect_uri` value.
```

Comment:
I'm not sure the 'MAY' clause is technically a restriction so a different word might be better. I'm struggling to come up with a really good word but 'clauses' may be better?

```suggestion
* `redirect_uri`: This value indicates that the Verifier's redirect URI is also the value of the Client Identifier. In this case, the Authorization Request MUST NOT be signed. The Verifier MAY omit the `redirect_uri` Authorization Request parameter. All Verifier metadata parameters MUST be passed using the `client_metadata` or `client_metadata_uri` parameter defined in (#vp_token_request). When Response Mode `direct_post` is used, these clauses apply to the `response_uri` value instead of `redirect_uri` value.
```

Reply:
omitting something is a technical restriction, no?

Reply:
I'm not sure. I think allowing someone to optionally omit something is technically an addition rather than a restriction. [addition probably isn't quite the right word, hopefully the point still makes sense :) ]

I didn't want to give anyone any wiggle room to argue they they don't have to allow verifiers to omit response_uri.